### PR TITLE
Feature/modify hours reports

### PR DESCRIPTION
**Pull request by AndreyEliVar**
https://ingesoftg001.atlassian.net/browse/PL-25
Complete annotations in the add hours form

### CODE REVIEW

File: pages/hours.js [hunk 1]
```diff
@@ -111,25 +114,45 @@ const AddHoursEmployee = ({
...
     const projects = proyectString;
+    let hoursUsers = [];
+    for (let hour of hoursWithId) {
+        hoursUsers.push(hour);
+    }
+    const [hourToAdd, setHourToAdd] = useState([]);
+    const [nextId, setNextId] = useState(hoursUsers.length);
     const [showModal, setShowModal] = useState(false);
     const [hours, setHoursState] = useState(0);
     const [date, setDateState] = useState("2022-10-08 00:00:00");
```
Comment (daar2510) [inline pages/hours.js:125]: Date initial value should be new Date() so it defaults to today's date

File: pages/hours.js [hunk 2]
```diff
@@ -46,51 +46,54 @@ export async function getServerSideProps(context) {
...
     projectQuery = projectQuery.filter((project) =>
         employeeProjects.some(
             (employee) => employee.nombreProyecto === project.nombre
         )
     );
 
     const proyectString = JSON.parse(safeJsonStringify(projectQuery));
+    proyectString.push({
+        nombre: "Mostrar todos",
+    });
```
Comment (daar2510) [inline pages/hours.js:96]: Please rename to "Todos los proyectos" like in the other pages.

File: pages/hours.js (hunk 1 above)
Comment (daar2510) [inline pages/hours.js:125]: Date initial value should be new Date() so it defaults to today's date

File: pages/hours.js [hunk 3]
```diff
@@ -111,25 +114,45 @@ const AddHoursEmployee = ({
...
+    const [nextId, setNextId] = useState(hoursUsers.length);
     const [showModal, setShowModal] = useState(false);
     const [hours, setHoursState] = useState(0);
     const [date, setDateState] = useState("2022-10-08 00:00:00");
     const [button, setButtonState] = useState(true);
     const [searchText, setSearchText] = useState("");
-    const [selectedProjectName, setSelectedProjectName] = useState("");
+    const [selectedProjectName, setSelectedProjectName] = useState(
+        projects[0] ? projects[0].nombre : ""
+    );
```
Comment (daar2510) [inline pages/hours.js:130]: Default value should be "Todos los proyectos" instead of the first project.

File: pages/hours.js [hunk 4]
```diff
@@ -176,10 +199,25 @@ const AddHoursEmployee = ({
...
-        router.reload();
+
+        if (response.status === 200) {
+            const newHour = {
+                id: nextId,
+                hours: parseInt(hours),
+                date: date
+                    .substring(0, 10)
+                    .concat(" | ")
+                    .concat(date.substring(11, 19)),
```
Comment (daar2510) [inline pages/hours.js:210]: Dont include seconds in the hour, e.g. 2:30 instead of 2:30:00

**Review by daar2510 — CHANGES_REQUESTED** [review]
Fix hour format
Fix dropdown options
Fix default date when modal is opened
Period validations are missing: registering hours of future days is not allowed and neither is registering hours in dates earlier than project's start date
Also I noticed you have unused variables in many files. Some regarding the now deleted searchbar and some unused useStates.

File: components/EmptyProjects.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+             {'Aún no tienes proyectos con contratos por horas.'}
+            </TextoEntrada>
+         </Text>
+         <UndrawPartyingReAt7F  src={"https://res.cloudinary.com/dwb71vld4/image/upload/v1657911038/Pollolandia/Logo_xjtzio.svg"} alt={"welcome"}/>
+       </HeroContent>
+     </Hero1>
+   );
+}
+
+export default EmptyProjects;
```
Comment (julian-solisaguilar) [inline components/EmptyProjects.js:95]: Add EOF

**Review by daar2510 — APPROVED** [review]
🚀🚀

**Pull request merged**